src/Raven.Server/Documents/Sharding/ShardedContext.cs

```diff
@@ -27,6 +28,7 @@ public ShardedContext(ServerStore server, DatabaseRecord record)
             {
                 var allNodes = server.GetClusterTopology().AllNodes;
                 var urls = record.Shards[i].AllNodes.Select(tag => allNodes[tag]).ToArray();
+                // TODO: pool request executors?
```

Comment:
Don't think we need to. The context is shared among all calls, no?

Reply:
Yes, you are correct.
So we need to update the database record upon each request, I think (or make it lazy?).

Reply:
What? Why update it per request?

Reply:
I might rely on the recent state of the database record. Right now we materializing it once.. and that's it.

Reply:
IIRC, this is updated whenever the database record is.